code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/projects/ProjectMemberResourceManager.java

```diff
@@ -55,14 +55,23 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
         Policy policy = (Policy) ApiContext.getContext().getPolicy();
         String id = RequestUtils.makeSingularStringIfCan(criteria.get("id"));
         if (StringUtils.isNotEmpty(id)) {
-            ProjectMember projectMember = authDao.getProjectMember(Long.valueOf(id));
+            ProjectMember projectMember;
+            try {
+                 projectMember = authDao.getProjectMember(Long.valueOf(id));
+            } catch (NumberFormatException e) {
+                return null;
```

Comment:
In case of garbage, probably raise NOT_FOUND. Theoretically more secure since malicious users would not be able to differentiate between a legit id that isnt there and a garbage id.